src/SFA.DAS.ProviderApprenticeshipsService.Web/Controllers/BulkUploadController.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web.Mvc;
+
+using SFA.DAS.ProviderApprenticeshipsService.Web.Models;
+using SFA.DAS.ProviderApprenticeshipsService.Web.Models.BulkUpload;
+using SFA.DAS.ProviderApprenticeshipsService.Web.Orchestrators;
+
+namespace SFA.DAS.ProviderApprenticeshipsService.Web.Controllers
+{
+    [RoutePrefix("{providerId}/apprentices")]
+    public class BulkUploadController : Controller
+    {
+        private readonly BulkUploadOrchestrator _bulkUploadController;
+
+        public BulkUploadController(BulkUploadOrchestrator bulkUploadOrchestrator)
+        {
+            if (bulkUploadOrchestrator == null)
+                throw new ArgumentNullException(nameof(bulkUploadOrchestrator));
+            _bulkUploadController = bulkUploadOrchestrator;
+        }
+
+        [HttpGet]
+        [Route("{hashedCommitmentId}/UploadApprenticeships")]
+        public async Task<ActionResult> UploadApprenticeships(long providerid, string hashedcommitmentid)
+        {
+            var viewModel = await _bulkUploadController.GetUploadModel(providerid, hashedcommitmentid);
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        [Route("{hashedCommitmentId}/UploadApprenticeships")]
+        public async Task<ActionResult> UploadApprenticeships(UploadApprenticeshipsViewModel uploadApprenticeshipsViewModel)
+        {
+            var result = await _bulkUploadController.UploadFile(uploadApprenticeshipsViewModel);
+            if (!result.Errors.Any())
+            {
+                // ToDo: Flash message, or other feedback to customer
+                return RedirectToAction("Details", "Commitment", new { uploadApprenticeshipsViewModel.ProviderId, uploadApprenticeshipsViewModel.HashedCommitmentId });
+            }
+
+            TempData["errors"] = result.Errors.ToList();
```

Comment:
We probably need to be careful here. As far as I'm aware TempData uses Session to store the data. In Prod it will be a multi-server farm so need to have session storage configured to handle this.